File: RQ1_Test_Generation/OpenAI_Data/HumanEvalPython_output/original/id_16T2000Test.py
# id_16_Test.java
from original.id_16 import count_distinct_characters

import pytest

class id_16_Test:
    """
    Test class of id_16.py.
    It contains ten unit test cases for the :func:`original.id_16.count_distinct_characters`.
    """


    def test_count_distinct_characters_1(self):
        """
        Test case 1 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('xyzXYZ') == 3

    def test_count_distinct_characters_2(self):
        """
        Test case 2 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('Jerry') == 4

    def test_count_distinct_characters_3(self):
        """
        Test case 3 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('Tom') == 2

    def test_count_distinct_characters_4(self):
        """
        Test case 4 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyz') == 26

    def test_count_distinct_characters_5(self):
        """
        Test case 5 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('ABCDEFGHIJKLMNOPQRSTUVWXYZ') == 26

    def test_count_distinct_characters_6(self):
        """
        Test case 6 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ') == 26

    def test_count_distinct_characters_7(self):
        """
        Test case 7 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789') == 36

    def test_count_distinct_characters_8(self):
        """
        Test case 8 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+-=[]{}|;:,.<>/?`~') == 70

    def test_count_distinct_characters_9(self):
        """
        Test case 9 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+-=[]{}|;:,.<>/?`~') == 70

    def test_count_distinct_characters_10(self):
        """
        Test case 10 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+-=[]{}|;:,.<>/?`~') == 70

# id_16_Test.py
from original.id_16 import count_distinct_characters

import pytest

class id_16_Test:
    """
    Test class of id_16.py.
    It contains ten unit test cases for the :func:`original.id_16.count_distinct_characters`.
    """
		
    def test_count_distinct_characters_1(self):
        """
        Test case 1 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('xyzXYZ') == 3

    def test_count_distinct_characters_2(self):
        """
        Test case 2 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('Jerry') == 4

    def test_count_distinct_characters_3(self):
        """
        Test case 3 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('Tom') == 2

    def test_count_distinct_characters_4(self):
        """
        Test case 4 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyz') == 26

    def test_count_distinct_characters_5(self):
        """
        Test case 5 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('ABCDEFGHIJKLMNOPQRSTUVWXYZ') == 26

    def test_count_distinct_characters_6(self):
        """
        Test case 6 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ') == 26

    def test_count_distinct_characters_7(self):
        """
        Test case 7 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789') == 36

    def test_count_distinct_characters_8(self):
        """
        Test case 8 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+-=[]{}|;:,.<>/?`~') == 70

    def test_count_distinct_characters_9(self):
        """
        Test case 9 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+-=[]{}|;:,.<>/?`~') == 70

    def test_count_distinct_characters_10(self):
        """
        Test case 10 for :func:`original.id_16.count_distinct_characters`.
        """
        assert count_distinct_characters('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+-=[]{}|;:,.<>/?`~') == 70

# id_17.java

/**
 * Given a string, find out how many distinct characters (regardless of case) does it consist of
 * @param string
 * @return
 */
public static int countDistinctCharacters(String string) {
    return (int) string.chars().map(Character::toLowerCase).distinct().count();
}

# id_17.py


def count_distinct_characters(string: str) -> int:
    """ Given a string, find out how many distinct characters (regardless of case) does it consist of
    >>> count_distinct_characters('xyzXYZ')
    3
    >>> count

File: RQ2_Prompt_Elements/GPT3.5_Data/HumanEvalJava_output/scenario3/id_16T2000Test.java
// CountDistinctCharacters.java
package scenario3;

import java.util.HashSet;
import java.util.Set;

class CountDistinctCharacters {

    /**
     * Given a string, find out how many distinct characters (regardless of case)  it consists  of.
     *
     * > countDistinctCharacters("xyzXYZ")
     * 3
     * > countDistinctCharacters("Jerry")
     * 4
     * > countDistinctCharacters("")
     * 0
     * > countDistinctCharacters("abcde")
     * 5
     * > countDistinctCharacters("abcdecadeCADE")
     * 5
     * > countDistinctCharacters("aaaaAAAAaaaa")
     * 1
     * > countDistinctCharacters("Jerry jERRY JeRRRY")
     * 5
     */
    public static int countDistinctCharacters(String string) {
        Set<Character> distinctChars = new HashSet<>();
        for (char c : string.toCharArray()) {
            if (Character.isLetter(c)) {
                distinctChars.add(Character.toLowerCase(c));
            }
        }
        return distinctChars.size();
    }
}

// CountDistinctCharactersTest.java
package scenario3;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link CountDistinctCharacters}.
* It contains ten unit test cases for the {@link CountDistinctCharacters#countDistinctCharacters(String)} method.
*/
class CountDistinctCharactersTest {

    @Test
    void testEmptyString() {
        assertEquals(0, CountDistinctCharacters.countDistinctCharacters(""));
    }

    @Test
    void testSingleDistinctCharacter() {
        assertEquals(1, CountDistinctCharacters.countDistinctCharacters("a"));
    }

    @Test
    void testSingleNonLetterCharacter() {
        assertEquals(0, CountDistinctCharacters.countDistinctCharacters("1"));
    }

    @Test
    void testMultipleDistinctCharacters() {
        assertEquals(5, CountDistinctCharacters.countDistinctCharacters("abcde"));
    }

    @Test
    void testMultipleMixedCaseDistinctCharacters() {
        assertEquals(5, CountDistinctCharacters.countDistinctCharacters("abcdecadeCADE"));
    }

    @Test
    void testMultipleSameCharacters() {
        assertEquals(1, CountDistinctCharacters.countDistinctCharacters("aaaaAAAAaaaa"));
    }

    @Test
    void testMultipleMixedCaseSameCharacters() {
        assertEquals(1, CountDistinctCharacters.countDistinctCharacters("AAAAaaaa"));
    }

    @Test
    void testMultipleWords() {
        assertEquals(5, CountDistinctCharacters.countDistinctCharacters("Jerry jERRY JeRRRY"));
    }

    @Test
    void testSpecialCharacters() {
        assertEquals(2, CountDistinctCharacters.countDistinctCharacters("a!b@c#d$e%f^g&h*i(j)k_l+m=n{o}p[q]r\\s|t`u~v"));
    }

    @Test
    void testAllLetters() {
        assertEquals(26, CountDistinctCharacters.countDistinctCharacters("abcdefghijklmnopqrstuvwxyz"));
    }
}